### DISCUSSION TITLE
Save compiled nlmixr model to cache ?

### DISCUSSION THREAD

**Discussion by Claire-PYC** [nlmixrdevelopment/nlmixr#539] [Q&A]
Dear nlmixr developer team,

I am running posthoc nlmixr analyses within a Docker container.
At the moment, the first call to the analysis takes about 1 minute to run, while all subsequent runs (whether or not they use different event inputs, patient data, etc...) only take a fraction of this time. 
I am new to nlmixr and Docker environment but I was under the impression that the nlmixr model structure needs to be formatted or compiled the very first time it is called, and this step is not necessary to the analyses after that. 

If this is the correct interpretation, is it possible to somehow save this compiled state for later use when starting a new session, in order to improve performance ? I have been able to save model fit objects for instance (.rds), but that did not do the trick.

Many thanks,
Claire

**Comment by mattfidler** [comment]
Hi Claire,

You can use `rxCreateCache()` to create a cache that is shared between R sessions,  It is also what is used for nlmixr

**Reply by Claire-PYC** [reply]
Hi Matthew,
Thank you very much for your answer. 
We managed to save some computing time for the first analysis by compiling the model as part of the Docker build.

**Reply by mattfidler** [reply]
Great!  Thanks for sharing
